From b7ed2a862ddafa7dbaa6299ee8633b6f8221e283 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@kernel.dk>
Date: Thu, 9 Jan 2020 14:58:51 -0700
Subject: [PATCH 49/77] io_uring: set sqe iopriority, if prio/prioclass is set

Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 engines/io_uring.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/engines/io_uring.c b/engines/io_uring.c
index 7c19294..4f6a967 100644
--- a/engines/io_uring.c
+++ b/engines/io_uring.c
@@ -206,6 +206,10 @@ static int fio_ioring_prep(struct thread_data *td, struct io_u *io_u)
 		}
 		if (!td->o.odirect && o->uncached)
 			sqe->rw_flags = RWF_UNCACHED;
+		if (fio_option_is_set(&td->o, ioprio_class))
+			sqe->ioprio = td->o.ioprio_class << 13;
+		if (fio_option_is_set(&td->o, ioprio))
+			sqe->ioprio |= td->o.ioprio;
 		sqe->off = io_u->offset;
 	} else if (ddir_sync(io_u->ddir)) {
 		if (io_u->ddir == DDIR_SYNC_FILE_RANGE) {
-- 
1.8.3.1

